pkg/pdfcpu/write.go

```diff
@@ -31,23 +31,20 @@ import (
 )
 
 // Write generates a PDF file for the cross reference table contained in Context.
-func Write(ctx *Context) error {
-
-	var file *os.File
-	var err error
-
+func Write(ctx *Context) (err error) {
 	// Create a writer for dirname and filename if not already supplied.
 	if ctx.Write.Writer == nil {
 
 		fileName := filepath.Join(ctx.Write.DirName, ctx.Write.FileName)
 		log.CLI.Printf("writing to %s\n", fileName)
 
-		file, err = os.Create(fileName)
+		file, err := os.Create(fileName)
 		if err != nil {
 			return errors.Wrapf(err, "can't create %s\n%s", fileName, err)
 		}
 
 		ctx.Write.Writer = bufio.NewWriter(file)
+		ctx.Write.Fp = file
```

Comment:
I think that we can safely override Fp while Writer is nil.

Reply:
We don't really need this, do we?